app/src/main/res/layout/dialog_critical_app_update.xml

```diff
@@ -0,0 +1,107 @@
+    android:textAppearance="?attr/textAppearanceBody1"
+    android:textColor="?attr/colorError"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toBottomOf="@id/criticalUpdateDescription" />
+
+  <com.google.android.material.button.MaterialButton
+    android:id="@+id/updateNowButton"
+    style="@style/Widget.Simple.Button.Red1"
+    android:layout_width="@dimen/spacing_0"
+    android:layout_height="wrap_content"
+    android:layout_marginTop="@dimen/spacing_32"
+    android:elevation="@dimen/spacing_2"
+    android:text="@string/critical_update_required_dialog_update_button_text"
+    android:textAppearance="?attr/textAppearanceButtonBig"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toBottomOf="@id/criticalUpdateReason" />
+
+  <View
+    android:id="@+id/divider"
+    android:layout_width="match_parent"
+    android:layout_height="@dimen/spacing_1"
+    android:layout_marginTop="@dimen/spacing_24"
+    android:background="#DADDE0"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toBottomOf="@id/updateNowButton" />
+
+  <com.google.android.material.textview.MaterialTextView
+    android:id="@+id/helpSectionTitle"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginTop="@dimen/spacing_24"
+    android:text="@string/critical_update_required_dialog_help_section_title"
+    android:textAppearance="?attr/textAppearanceBody0Medium"
+    android:textColor="?attr/colorOnSurface"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toBottomOf="@+id/divider" />
+
+  <com.google.android.material.textview.MaterialTextView
+    android:id="@+id/helpSectionDescription"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginTop="@dimen/spacing_16"
+    android:text="@string/critical_update_required_dialog_help_section_content"
+    android:textAppearance="?attr/textAppearanceBody0"
+    android:textColor="?attr/colorOnSurface"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toBottomOf="@+id/helpSectionTitle" />
+
+  <com.google.android.material.button.MaterialButton
+    style="@style/Widget.Simple.Button.Green2"
+    android:id="@+id/supportTeamContactButton"
+    android:layout_width="@dimen/spacing_0"
+    android:layout_height="wrap_content"
+    android:layout_marginTop="@dimen/spacing_16"
+    android:elevation="@dimen/spacing_2"
+    android:textAppearance="?attr/textAppearanceButtonBig"
+    app:elevation="@dimen/spacing_4"
+    app:icon="@drawable/ic_whatsapp_logo"
+    app:iconTint="?attr/colorSecondary"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toBottomOf="@id/helpSectionDescription" />
```

Comment:
This is good. But do you think we can move the icon, text, and icon tint to the style itself? 

For example, we can have a style named

```xml
  <style name="Widget.Simple.Button.WhatsApp">
```

and similarly `Telegram`, and so on. That way we can just update the style programmatically WDYT? 

_We can do this in the next PR_

Reply:
Afaiu we can not update the style of a button dynamically. There is no `.setStyle` method available. Correct me if I'm wrong? Seems like having two buttons is a better option after all. Thoughts? @msasikanth

Reply:
We would have to go with 3 different buttons, WhatsApp, Telegram, and a default fallback button for backward compatibility. We cannot update the style directly, but we can take a look at updating the style using a `[ContextThemeWrapper](https://developer.android.com/reference/android/view/ContextThemeWrapper)` for example or updating the individual properties like background, icon, and text at runtime.